matron/src/device/device_monome.c

```diff
@@ -87,17 +86,39 @@ void dev_monome_all_led(struct dev_monome *md, uint8_t val) {
     }
 }
 
+// set a given arc LED value
+void dev_arc_set_led(struct dev_monome *md,
+                        uint8_t enc, uint8_t led, uint8_t val) {
+    /* fprintf(stderr, "dev_monome_set_led: %d %s %d %d %d\n", */
+    /*             md->dev.id, md->dev.serial, enc, led, val); */
+    md->data[enc][led] = val;
+    md->dirty[enc] = true;
+    md->arc = true;
+}
+
+// set all arc LEDs to value
+void dev_arc_all_led(struct dev_monome *md, uint8_t val) {
+    dev_monome_all_led(md, val);
+    md->arc = true;
```

Comment:
this flag shouldn't be set here or in the above method. i'd suggest removing it completely and using rows/cols for checking that we're dealing with an arc in `dev_monome_refresh` (arcs have 0x0 size in libmonome)

Reply:
i'm considering implementing separate arc methods instead of re-using `dev_monome_refresh`/`dev_monome_init` etc. it wouldn't create much code duplication and it would be cleaner (and i have a feeling this would help with the issue of grid and arc not working together).

Reply:
this will also make things a bit messier. i believe the issue with grid/arc not working together could be solved with the current implementation. could you run gdb to see what leads to the memory errors?

Reply:
wouldn't it be cleaner with separate methods?

`void dev_arc_all_led(...)`
`void dev_grid_all_led(...)`
vs
```
void dev_monome_all_led(...)
...
if (is_arc)
    // arc implementation
else
  // grid implementation
```
they are pretty light weight wrappers, so there will be hardly any code duplication (and if there is it could be extracted into generic monome functions which both arc and grid functions would call).

regardless yeah, i'll investigate the problem some more, i have a feeling we might run into this with multiple grids as well. thanks for the suggestion, will give it a try!